heartbeat/minio

```diff
@@ -0,0 +1,275 @@
+
+        ocf_log debug "Starting minio: $cmd"
+
+        eval $cmd > ${OCF_RESKEY_pidfile}
+
+        if [ "$?" -ne 0 ]; then
+                ocf_log err "Minio returned error" $?
+                exit $OCF_ERR_GENERIC
+        fi
+
+        exit $OCF_SUCCESS
+}
+
+
+minio_stop()
+{
+        if minio_status ; then
+                PID=`head -n 1 $OCF_RESKEY_pidfile`
+                if [ ! -z "$PID" ]; then
+                        ocf_log info "Killing Minio PID $PID"
+                        kill $PID > /dev/null 2>&1
+                        if [ "$?" -eq 0 ]; then
+                                TRIES=0
+                                while isRunning "$PID" && [ "$TRIES" -lt 30 ]
+                                do
+                                        sleep 1
+                                        ocf_log info "Minio PID $PID is still running"
+                                        TRIES=`expr $TRIES + 1`
+                                done
+                                isRunning "$PID"
+                                RET=$?
+                                if [ "$RET" -eq 0 ]; then
+                                        ocf_log info "Killing Minio PID $PID with SIGKILL"
+                                        kill -9 $PID > /dev/null 2>&1
+                                        while isRunning "$PID"
+                                        do
+                                                sleep 1
+                                                ocf_log info "Minio PID $PID is still running"
+                                        done
+                                fi
+                        else
+                                ocf_log err "Killing Minio PID $PID FAILED"
+                                exit $OCF_ERR_GENERIC
+                        fi
+                fi
+        fi
+
+        exit $OCF_SUCCESS
+}
+
+minio_monitor()
+{
+        minio_status
+        RET=$?
+
+        if [ "$RET" -eq 0 ]; then
+                PID=`head -n 1 $OCF_RESKEY_pidfile`
+                ocf_log debug "Minio monitor on PID $PID succeeded"
+                return $OCF_SUCCESS
+        else
+                ocf_log debug "Minio monitor on PID $PID failed"
+                return $OCF_NOT_RUNNING
+        fi
+}
+
+minio_validate_all()
+{
+
+        # check that the minio binary exists
+        if [ ! -x "$OCF_RESKEY_binary" ]; then
+                ocf_log err "Minio server binary $OCF_RESKEY_binary does not exist"
+                exit $OCF_ERR_INSTALLED
+        fi
+
+        # check that the Minioconfig file exists
+        if [ ! -d "$OCF_RESKEY_confdir" ]; then
+                ocf_log err "Minio config dir $OCF_RESKEY_confdir does not exist"
+                exit $OCF_ERR_CONFIGURED
+        fi
+
+}
+
+#
+# Main
+#
+
+if [ $# -ne 1 ]
+then
+  usage
+  exit $OCF_ERR_ARGS
+fi
+
+case $1 in
+    start)      minio_validate_all
+                        minio_start
```

Comment:
Cleanup in indentation here would help making this case-statement easier to read.